style: numbered lines for code blocks

dvdksn opened the pull request:
Signed-off-by: David Karlsson <35727626+dvdksn@users.noreply.github.com>

Adds support for numbered lines in code blocks

<img width="805" alt="image" src="https://github.com/docker/docs/assets/35727626/aa790c29-ee40-4e51-8109-1209114738fd">

Syntax:

````
```go {linenos=true}
// code
```

OR

```go {linenos=table}
// code
```

OR

```go {linenos=inline}
// code
```
````

aevesdocker reviewed (changes requested):
Love this! <3
However...
For non-formatted code blocks, the formatting has changed and is looking a lil weird.
![Screenshot 2023-10-30 at 09 34 27](https://github.com/docker/docs/assets/102604716/cb3abd1b-bafe-4e9a-9905-1953c87ee99b)

![Screenshot 2023-10-30 at 09 34 15](https://github.com/docker/docs/assets/102604716/72b4a1e8-4ea4-4670-80fb-0b5f75ec0495)

dvdksn commented:
> formatting has changed and is looking a lil weird.
fixed, I think (couldn't reproduce exactly)

Also fixed the copy code button and added a no-select for prompt chars (unrelated change but improves copy behavior for single lines in console blocks)

aevesdocker reviewed (approved):
LGTM!
